src/adu_types/src/adu_types.c

```diff
@@ -9,6 +9,25 @@
 #include <stdlib.h>
 #include <string.h>
 
+/**
+ * @brief Returns the string associated with @p connType
+ * @param connType ADUC_ConnType to be stringified
+ * @returns if the ADUC_ConnType exists then the string version of the value is returned, "" otherwise
+ */
+const char* ADUC_ConnType_ToString(const ADUC_ConnType connType)
+{
+    switch (connType)
+    {
+    case ADUC_ConnType_NotSet:
+        return "ADUC_ConnType_NotSet";
+    case ADUC_ConnType_Device:
+        return "ADUC_ConnType_Device";
+    case ADUC_ConnType_Module:
+        return "ADUC_ConnType_Module";
+    }
+
+    return "";
```

Comment:
>"" [](start = 11, length = 2)

what do I do elsewhere? Do I send "n/a", or fail (e.g. return null)?  Returning "" can be confusing as if there's a bug it wont be obvious in trace what the value is.

Reply:
We normally use "<Unknown>"

I'll change this to that.

---
In reply to: [594760708](https://github.com/Azure/iot-hub-device-update/pull/16#discussion_r594760708) [](ancestors = 594760708)